### PR TITLE
use golang 1.21 until runc works with 1.22

### PR DESCRIPTION
**Pull request by kannon92**
#### What this PR does / why we need it:
/kind cleanup
#### Special notes for your reviewer:
go.mod is still set to 1.21.

Runc fails to build on golang 1.22 at the moment.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by kannon92** [comment]
cc @sohankunkerkar @kwilczynski @haircommander

**Review by saschagrunert — APPROVED** [review]

**Comment by kannon92** [comment]
/hold cancel
@saschagrunert PTAL.

The make verify-dependencies step was failing due to the github workflows using golang 1.22.

File: .github/workflows/test.yml
Comment (saschagrunert) [inline .github/workflows/test.yml:12]: We need go 1.22 for Kubernetes 1.30, which is scheduled for April 17.

Comment (kannon92) [reply .github/workflows/test.yml:12]: yea..

I'm not sure what to do here with runc not building on golang 1.22. Maybe we can see about runc relaxing the restriction on build as we are going to get it working with golang 1.22 eventually?

Comment (haircommander) [reply .github/workflows/test.yml:12]: opened https://github.com/kubernetes/kubernetes/issues/124168 as well as @saschagrunert and I brought up some slack threads with the release team

Comment (saschagrunert) [reply .github/workflows/test.yml:12]: Let's get this one in in the meanwhile.

**Comment by kannon92** [comment]
@saschagrunert @haircommander can I get a lgtm?

**Comment by kannon92** [comment]
https://github.com/cri-o/cri-o/pull/7956#issuecomment-2034807224

ha. nvm

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**
